_src/pattern-library/sass/components/_forms.scss

```diff
@@ -0,0 +1,455 @@
+    line-height: 100%; // centers the text vertically while matching height of replacement menu
+    z-index: z-index(base);
+
+    svg {
+        @include size(rem($baseline));
+        position: absolute;
+        right: rem(($baseline/2));
+        fill: $input-border-default;
+    }
+
+    &:focus,
+    &.is-active {
+        border: 1px solid $input-border-default-focus;
+
+        svg {
+            @include transition(all timing(xx-fast) ease-in-out 0s);
+            fill: $input-border-default-focus;
+        }
+    }
+
+    &:disabled,
+    &.is-disabled {
+        color: $input-text-disabled;
+        cursor: not-allowed;
+    }
+}
+
+// ----------------------------
+// #RADIO / CHECKBOX
+// ----------------------------
+.wrapper-replace-radio,
+.wrapper-replace-checkbox {
+    display: inline-block;
+    position: relative;
+    width: 1rem;
+}
+
+.wrapper-custom-radio,
+.wrapper-custom-checkbox {
+    display: inline-block;
+    z-index: z-index(base);
+
+    &.is-selected {
+
+    }
+}
+
+.input-radio,
+.input-checkbox {
+
+    &:disabled,
+    &.is-disabled {
+
+    }
+
+    &.replace-checkbox,
+    &.replace-radio {
+        @extend %a11y-hide;
+
+        & + .input-label:before {
+            @include size(rem($baseline/1.25));
+            display: inline-block;
+            margin-right: spacing-horizontal(x-small);
+            content: "";
+            vertical-align: top;
+        }
+    }
+
+    &.replace-checkbox {
+
+        & + .input-label:before {
+            @include transition(all timing(xx-fast) ease-in-out 0s);
+            border: 1px solid $input-border-default;
+            background: $input-check-radio-background-default;
+        }
+
+        &:checked + .input-label:before {
+            background: $input-check-radio-background-checked;
+            box-shadow: inset 0 0 0 2px palette(grayscale, white);
+        }
+    }
+
+    &.replace-radio {
+
+        & + .input-label:before {
+            @include transition(all timing(xx-fast) ease-in-out 0s);
+            border: 1px solid $input-border-default;
+            border-radius: 50%;
+            background: $input-check-radio-background-default;
+        }
+
+        &:checked + .input-label:before {
+            background: $input-check-radio-background-checked;
+            box-shadow: inset 00 0 0 2px palette(grayscale, white);
+        }
+    }
+}
+
+// ----------------------------
+// #TEXTAREA
+// ----------------------------
+.input-textarea {
+    width: input-width(base);
+    height: input-height(base);
+
+    &:focus,
+    &:active,
+    &.is-active {
+        border-color: $input-border-default-focus;
+        color: $input-text-default-hover;
+    }
+
+    &.textarea-short {
+        height: input-height(short);
+    }
+
+    &.textarea-tall {
+        height: input-height(tall);
+    }
+
+    &.textarea-block {
+        width: 100%;
+    }
+}
+
+// ----------------------------
+// #SLIDERS
+// ----------------------------
+.input-slider,
```

Comment:
Can you pull out the rules here so that its clear 1) what's a reset/cross-browser sync up vs. 2) Open edX/edX customization of the element?